### PR TITLE
Add genetic DE

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
current best performer on CompilerGym

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -1918,6 +1918,7 @@ def __init__(
 chainNaiveTBPSACMAPowell.no_parallelization = True
 
 # new names
+CGDE = Chaining([RotatedTwoPointsDE, TwoPointsDE], [200]).set_name("CGDE", register=True)
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:1921]: experimental variants for now?
and why not a more explicit name than an acronym?

Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py:1921]: It's the best performing algorithm on several benchmarks, and the main algorithm for reporting good results from one part of the genome to another. I would go for inclusion in the main optimizerlib.py
I update the name.

**Review by jrapin — APPROVED** [review]
ok then
i don't understand why "Genetic" though

**Comment by teytaud** [comment]
> ok then
> i don't understand why "Genetic" though

Genetic in the sense crossovers "a la Holland" make sense, even without restricting the cuts to be aligned... to be discussed offline, that's something cool and new in Nevergrad.

**Comment by teytaud** [comment]
> > ok then
> > i don't understand why "Genetic" though
> 
> Genetic in the sense crossovers "a la Holland" make sense, even without restricting the cuts to be aligned... to be discussed offline, that's something cool and new in Nevergrad.

You might think of "RotatedTwoPointsDE" as the critical component justifying the word "genetic".

**Pull request merged**